__tests__/game_controller_test.js

```diff
@@ -7,7 +7,101 @@ describe('GameController', () => {
     gameController = new GameController;
   });
 
-  it('creates an instance of the Game class when intitialised'), () => {
-    expect(gameController.game).toEqual(Game.prototype)
-  }
+  it('creates an instance of the Game class when initialized', () => {
+    expect(typeof gameController.game === 'object').toEqual(true);
+  });
+
+  it('allows a player to make a move', () => {
+    gameController.takeTurn(3)
+    expect(gameController.game.grid).toEqual([1, 2, 'x', 4, 5, 6, 7, 8, 9]);
+  });
+
+  it('allows a second player to make a move', () => {
+    gameController.takeTurn(3)
+    gameController.togglePlayer()
+    gameController.takeTurn(5)
+    expect(gameController.game.grid).toEqual([1, 2, 'x', 4, 'o', 6, 7, 8, 9]);
+  });
+
+  it('returns "Tie" when neither player has won after 9 moves', () => {
```

Comment:
Sorry to sound like a broken record, but dependency injection could avoid the need to do this! 

So if the Board constructor takes a grid, the Game takes a board, and the Game Controller takes a Game, we could make this a bit smaller! 

So

board.js
```
class Board
constructor(grid) {
    this.grid = grid;
    this.moves = 0;
    this.winningCombos = [
      [0, 1, 2],
      [3, 4, 5],
      [6, 7, 8],
      [0, 3, 6],
      [1, 4, 7],
      [2, 5, 8],
      [2, 4, 6],
      [0, 4, 8]
    ];
    this.checkCombos = false;
  }
```

game.js
```
class Game {
  constructor(board) {
    this.board = board;
    this.player = new Player;
    this.grid = this.board.grid;
    this.player1 = this.player.player1;
    this.player2 = this.player.player2;
    this.currentPlayer = this.player1;
  }
```

GameController.js
```
class GameController {
  constructor(game) {
    this.game = game;
  }
```

So to do this exact same test, we could write it as 

```
it('returns "Tie" when neither player has won after 9 moves', () => {
const board = new Board(['o', 'x', 'x', 'x', 'o', 'o', 'o', 'x', 'x']);
const game = new Game(board);
const gameConstructor = new GameConstructor(game);

expect(gameControlller.isATieOrWon()).to Equal("Tie");
}) 
```

Reply:
Thanks Dan. This was super helpful and we've refactored our classes today to include dependency injection